app/models/user.rb

```diff
@@ -24,9 +8,16 @@ def self.from_omni_auth(auth_hash)
     ) do |user|
       user.name = auth_hash["info"]["name"]
       user.image_url = auth_hash["info"]["image"]
-      user.url = auth_hash["info"]["urls"]["Twitter"]
-      user.save!
+      user.save
     end
     user
   end
+
+  def self.top_users
+    ActiveRecord::Base.connection.execute(
+      "SELECT users.name, users.image_url, sum(links.clicks) as total_clicks" \
+      " FROM links INNER JOIN users ON links.user_id = users.id" \
+      " GROUP BY users.id ORDER BY total_clicks DESC LIMIT 10"
+    )
+  end
```

Comment:
#### Code Quality - Critical

Stick to proper OOP methodology. Calculate top users from the database instead of storing a `total clicks` counter on the user model.